tools/ci_ensure_cleanup.ps1

```diff
@@ -0,0 +1,35 @@
+#
+# Copyright (c) .NET Foundation and contributors. All rights reserved.
+# Licensed under the MIT license. See LICENSE file in the project root for full license information.
+#
+
+<#
+.SYNOPSIS
+    This script is run before each CI run to ensure things are properly cleaned up
+.DESCRIPTION
+    This script is run before each CI run to ensure things are properly cleaned up
+.PARAMETER dropLocation
+
+#>
+
+# If two tests are run too closely, VBCSCompiler process might still be alive for previous test preventing dotnet and packages to be cleand up
+function KillVBCSCompilers() {
+    $procs = Get-WmiObject Win32_Process | Where-Object { $_.Name.ToLower().StartsWith("dotnet") -and $_.CommandLine.ToLower().Contains("vbcscompiler") }
+    foreach ($proc in $procs) {
+        Stop-Process -Id $proc.ProcessId -Force
+    }
+}
+
+# Some failed test would actually cause appHost config file to become corrupted and w3svc would not be able to start
+function EnsureAppHostConfig() {
+    $appConfigLocation = [System.IO.Path]::Combine($env:WinDir, "System32", "inetsrv", "config", "applicationHost.config")
+    if (!(Test-Path $appConfigLocation) -or ((Get-Item $appConfigLocation).Length -eq 0)) {
+        $appConfigBackupLocation = "${appConfigLocation}.ancmtest.masterbackup"
+        Copy-Item -Path $appConfigBackupLocation -Destination $appConfigLocation -Force

```

Comment:
After copying the file, you should double-check the file size is the same to the backup file. 
In some cases, the applicationhost.config file can be locked by other process and the copy-item will fail with throwing some exception error.
So, I'd like to suggest to add a exception handling for the copy-item and restart IIS services to force to unlock the file handle lock if the file size is not matched and try again.

Reply:
Good point. Maybe I should just fail the entire script at this point and block the CI build.
We can come back in and rerun the test or login to CI machine to manually fix the problem.